flyteidl/protos/flyteidl/core/tasks.proto

```diff
@@ -84,6 +84,14 @@ message RuntimeMetadata {
 
     //+optional It can be used to provide extra information about the runtime (e.g. python, golang... etc.).
     string flavor = 3;
+
+    //+optional It can be used to provide extra information for the plugin.
+    PluginMetadata plugin_metadata = 4;
+}
+
+message PluginMetadata {
+    //+optional It can be used to decide use sync plugin or async plugin during runtime.
```

Comment:
we should move it to agent proto

Reply:
I think it makes sense to be here? This is not strict only for agent, and it could be extended in the future to include more fields for other purposes. Also moving this to agent proto will result in circular imports that we will need to resolve somehow.

Reply:
@honnix why should the is_sync be a feature/property of the task rather than the agent/plugin?

Reply:
Sorry I have been able to follow up recent changes. This was needed for routing https://github.com/flyteorg/flyte/pull/4107/files#diff-da53a6f262afa680bad44d5bc3e5a6a95f1e8b8764002e0db9a7f015af7d2831R137-R154. Not sure whether this is still valid.

Reply:
It's hard for me to see the full picture, especially combined with #4511 . I have a feeling that metadata service is used to have a reflection type of thing so the routing here won't be needed, but I'm not entirely sure.

It would be great if you can outline the overall architecture of these changes.